CMakeLists.txt

```diff
@@ -180,6 +180,10 @@ if (NOT "cxx_std_${CMAKE_CXX_STANDARD}" IN_LIST ROOT_COMPILE_FEATURES)
   message(WARNING "ROOT has not been built with the requested C++ standard ${CMAKE_CXX_STANDARD}.")
 endif()
 
+# TORCH
+list(APPEND CMAKE_PREFIX_PATH "/opt/software/linux-debian12-x86_64_v2/gcc-12.2.0/py-torch-2.1.0-wdoylpbmycgoq2fq4upnnx3tli4335mv/lib/python3.10/site-packages/torch/share/cmake/Torch")
```

Comment:
This should become a PR for py-torch in slack because https://github.com/spack/spack/blob/4022f083d5c2f770f83397729d6371b6012444a0/var/spack/repos/builtin/packages/py-torch/package.py#L642 should work I think.

Reply:
This guy claims it should work. I wouldn't trust him though. https://github.com/spack/spack/pull/37012#issuecomment-1515408329

Reply:
Also, https://github.com/spack/spack/issues/38265, maybe Thomas has gotten this working? Could check at next key4hep meeting or on their mattermost.

Reply:
If I remove this line, I get ```
CMake Error at CMakeLists.txt:184 (find_package):
  By not providing "FindTorch.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Torch", but
  CMake did not find one.

  Could not find a package configuration file provided by "Torch" with any of
  the following names:

    TorchConfig.cmake
    torch-config.cmake

  Add the installation prefix of "Torch" to CMAKE_PREFIX_PATH or set
  "Torch_DIR" to a directory containing one of the above files.  If "Torch"
  provides a separate development package or SDK, be sure it has been
  installed.
```

Reply:
I know. But that path is not stable. It will change on a timescale of once a week due to container upgrades etc.

Reply:
Try
```suggestion
list(APPEND CMAKE_PREFIX_PATH "/usr/local/lib/python3.10/site-packages/torch/share/cmake/Torch")
```

Reply:
Do we need to modify eic-spack packages yaml file? 

```
jug_xl> rahmans@ifarm1801:/w/eic-scshelf2104/users/rahmans/RomanPotsML/EICrecon$ echo $CMAKE_PREFIX_PATH
/usr/local
```

Reply:
Maybe add as `PATHS` in find_package, https://cmake.org/cmake/help/latest/command/find_package.html

Reply:
Like this? Wouldn't this break as soon as pytorch is upgraded in the container. 
`find_package(Torch REQUIRED PATHS "/opt/software/linux-debian12-x86_64_v2/gcc-12.2.0/py-torch-2.1.0-wdoylpbmycgoq2fq4upnnx3tli4335mv/lib/python3.10/site-packages/torch/share/cmake/Torch")
`

Reply:
Yeah, still with /usr/local.

Reply:
Moved the torch finding to the jana.cmake file https://github.com/eic/EICrecon/pull/1207/commits/14cb1002b31d5c66471f3adc9a0c0ccf85eee64d

Reply:
Perhaps spack should be putting related paths to the CMAKE_PREFIX_PATH

Reply:
That's what all the linked PRs and issues are suggesting ;-)

One could also say that Torch should install CMake configs in the right place, not under the python site-packages...

Reply:
> Perhaps spack should be putting related paths to the CMAKE_PREFIX_PATH

If I understand correctly, that's what the spack PR mentioned above was supposed to do.